x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/aggs/InferencePipelineAggregationBuilder.java

```diff
@@ -0,0 +1,236 @@
+        out.writeOptionalNamedWriteable(inferenceConfig);
+        gapPolicy.writeTo(out);
+    }
+
+    @Override
+    protected PipelineAggregator createInternal(Map<String, Object> metaData) {
+
+        SetOnce<Model> model = new SetOnce<>();
+        SetOnce<Exception> error = new SetOnce<>();
+        CountDownLatch latch = new CountDownLatch(1);
+        ActionListener<Model> listener = new LatchedActionListener<>(
+            ActionListener.wrap(model::set, error::set), latch);
+
+        modelLoadingService.get().getModelForSearch(modelId, listener);
+        try {
+            // TODO Avoid the blocking wait
+            latch.await();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new RuntimeException("Inference aggregation interrupted loading model", e);
+        }
+
+        Exception e = error.get();
+        if (e != null) {
+            if (e instanceof RuntimeException) {
+                throw (RuntimeException)e;
+            } else {
+                throw new RuntimeException(error.get());
+            }
+        }
+
+        InferenceConfigUpdate update = adaptForAggregation(inferenceConfig);
+
+        return new InferencePipelineAggregator(name, bucketPathMap, metaData, update, model.get());
+    }
+
+    static InferenceConfigUpdate adaptForAggregation(InferenceConfigUpdate originalUpdate) {
+        InferenceConfigUpdate updated;
+        if (originalUpdate == null) {
+            updated = new ResultsFieldUpdate(AGGREGATIONS_RESULTS_FIELD);
+        } else {
+            if (originalUpdate instanceof ClassificationConfigUpdate) {
+                ClassificationConfigUpdate classUpdate = (ClassificationConfigUpdate)originalUpdate;
+
+                // error if the top classes result field is set and not equal to the only acceptable value
+                String topClassesField = classUpdate.getTopClassesResultsField();
+                if (Strings.isNullOrEmpty(topClassesField) == false &&
+                    ClassificationConfig.DEFAULT_TOP_CLASSES_RESULTS_FIELD.equals(topClassesField) == false) {
+                    throw ExceptionsHelper.badRequestException("setting option [{}] to [{}] is not valid for inference aggregations",
+                        ClassificationConfig.DEFAULT_TOP_CLASSES_RESULTS_FIELD, topClassesField);
+                }
+            }
+
+            // error if the results field is set and not equal to the only acceptable value
+            String resultsField = originalUpdate.getResultsField();
+            if (Strings.isNullOrEmpty(resultsField) == false && AGGREGATIONS_RESULTS_FIELD.equals(resultsField) == false) {
+                throw ExceptionsHelper.badRequestException("setting option [{}] to [{}] is not valid for inference aggregations",
+                    ClassificationConfig.RESULTS_FIELD.getPreferredName(), resultsField);
+            }
```

Comment:
These types of validations seem like they belong in the agg parser.

Maybe we should always parse the named objects into a `ResultsFieldUpdate` object or something. 

At a minimum, these validations should occur before we attempt to load the model.